### PR TITLE
[custom op] Use canonical API to constrain unbacked values

### PR DESCRIPTION
**Pull request by tugsbayasgalan**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #108372

### CODE REVIEW

**Review by angelayi — APPROVED** [review]

**Comment by ezyang** [comment]
Two thoughts: (1) are these intended to have the same semantics? If not, what is the intended semantic difference? (2) It seems strange for this function to live in `export` namespace if it is used in non-export cases.

**Comment by tugsbayasgalan** [comment]
> Two thoughts: (1) are these intended to have the same semantics? If not, what is the intended semantic difference? (2) It seems strange for this function to live in `export` namespace if it is used in non-export cases.

For 1) The previous behaviour is that no matter what user passes in as min value, it always assumes the min value is 2. This seems like a bug to me. With this new change, user can use any min/max value to constraint symints as long as they are compatible values according to constrain_as_size API. 

For 2) Yeah it does seem weird that these functions live in export namespace, they are more of a "tracing" concepts. I can use the underlying aten op directly.

**Comment by ezyang** [comment]
> For 1) The previous behaviour is that no matter what user passes in as min value, it always assumes the min value is 2. This seems like a bug to me. With this new change, user can use any min/max value to constraint symints as long as they are compatible values according to constrain_as_size API.

This doesn't quite answer my question though, which is SHOULD the apis be different?

**Comment by tugsbayasgalan** [comment]
> > For 1) The previous behaviour is that no matter what user passes in as min value, it always assumes the min value is 2. This seems like a bug to me. With this new change, user can use any min/max value to constraint symints as long as they are compatible values according to constrain_as_size API.
> 
> This doesn't quite answer my question though, which is SHOULD the apis be different?

Ohh haha hmm. From the docstring, it sounds like this API is used for tracing size-like unbacked symints. So maybe it makes sense to have same behaviour as constrain_as_size API. Or are you saying we should just use constrain_as_size instead of having a seperate API for custom op? cc: @zou3519

**Comment by ezyang** [comment]
we'll chat about this more in our 1:1, but my claim is (1) that the user facing constrain API doesn't need to actually be a real honest to goodness operator, because we should be inserting necessary runtime asserts anyway, (2) torch.sym_* is a horrible user facing API, (3) the constrain APIs shouldn't live in torch.export since they are broadly applicable

File: torch/_custom_op/impl.py [hunk 1]
```diff
@@ -888,15 +882,8 @@ def create_unbacked_symint(self, *, min=2, max=None) -> torch.SymInt:
...
-        if min < 2:
-            raise ValueError(
-                f"ctx.create_unbacked_symint(min={min}, ...): expected min to be "
-                f"greater than or equal to 2. PyTorch only supports new "
-                f"data-dependent sizes of >= 2"
-            )
-
         result = self._shape_env.create_unbacked_symint()
-        torch.fx.experimental.symbolic_shapes.constrain_range(result, min=2, max=max)
+        torch.sym_constrain_range_for_size(result, min=min, max=max)
```
Comment (ezyang) [inline torch/_custom_op/impl.py:886]: I don't think this will actually work, due to unrelated bugs

**Comment by ezyang** [comment]
this is up to @tugsbayasgalan

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**